### PR TITLE
Issue2

### PR DESCRIPTION
**Pull request by kdeng121**

### CODE REVIEW

**Comment by kdeng121** [comment]
[Issue 2: Save Image Link](https://trello.com/c/3LEMOJus/60-issue-2-save-image-link)

**Comment by kdeng121** [comment]
[Issue 2: Save Image Link](https://trello.com/c/22qytdPg/61-issue-2-save-image-link)

File: package-lock.json [hunk 1]
```diff
@@ -7,48 +7,48 @@
       "resolved": "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.0.0.tgz",
```
Comment (ztmeyer) [inline package-lock.json]: I think during our rebase we squashed these 2 commits together - probably not worth fixing but I wanted to point it out

File: test/models/image_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (ztmeyer) [inline test/models/image_test.rb]: Could we add some tests here?

While we're at it, could we add some ImageController tests? (add a `test/controllers/images_controller_test.rb` file)

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -3,5 +3,22 @@ def new
     @image = Image.new
   end
 
-  def create; end
+  def create
```
Comment (raj-al-ghul) [inline app/controllers/images_controller.rb]: did something happen to the commit message (`kd - Update package-lock.json`)?

File: test/models/image_test.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+class ImageTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (ztmeyer) [inline test/models/image_test.rb]: You can remove these comments

File: test/models/image_test.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    assert_not image.save, 'Saved image with blank URL'
+  end
+
+  test 'URL is valid' do
+    image = Image.new
+    image.url = 'test.com'
+    assert_not image.save
+    image.url = 'http://test.com'
+    assert image.save
+  end
```
Comment (ztmeyer) [inline test/models/image_test.rb]: If I'm testing validations I usually add a test that instantiates the object (`Image.new`), seeds it with values (urls) that are bad, then calls `.valid?` on the object for each value

File: test/controllers/images_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  test 'Create action accepts URL, saves and redirects' do
+    assert_difference 'Image.count', 1 do
+      post images_path,
+           params: { image: { url: 'http://test.com' } }
+    end
+    added_url = Image.last.url
+    assert_equal 'http://test.com', added_url
+    assert_response :redirect
+    follow_redirect!
```
Comment (ztmeyer) [inline test/controllers/images_controller_test.rb]: I haven't seen `follow_redirect!` before - looks pretty neat, but should we be doing that in a test? My sense is that we shouldn't

File: app/controllers/images_controller.rb [hunk 7]
```diff
@@ -8,7 +8,7 @@ def create
     if @image.save
       redirect_to @image
     else
-      render 'new'
+      render :new, status: :unprocessable_entity
```
Comment (raj-al-ghul) [inline app/controllers/images_controller.rb]: ideally we'd fix this up

File: test/controllers/images_controller_test.rb (hunk 6 above)
Comment (ztmeyer) [reply test/controllers/images_controller_test.rb]: Actually I think we can use `assert_redirected_to`

**Comment by raj-al-ghul** [comment]
good work! lets keep improving on cleaning up the commits going forward

**Pull request merged**